File: frontend/src/services/tracking.ts
import { api } from "@/lib/api";

export interface OrderTracking {
  order_id: number;
  order_number: string;
  status: string;
  driver_id: number | null;
  driver_name: string | null;
  driver_phone: string | null;
  driver_latitude: number | null;
  driver_longitude: number | null;
  driver_zone: string | null;
  driver_vehicle_type: "moto" | "bicicleta" | "auto" | null;
  driver_updated_at: string | null;
  delivery_address: string;
  is_active: boolean;
}

export async function getOrderTracking(orderId: number) {
  const { data } = await api.get<OrderTracking>(`/orders/${orderId}/tracking`);
  return data;
}

export async function updateDriverLocation(payload: {
  latitude: number;
  longitude: number;
  zone?: string;
}) {
  const { data } = await api.patch("/delivery/location", payload);
  return data;
}

File: mobile/lib/models/tracking.dart
import '../core/format.dart';
import '../models/order.dart';

/// Espejo del dict devuelto por GET /orders/{id}/tracking.
class OrderTracking {
  OrderTracking({
    required this.orderId,
    required this.orderNumber,
    required this.status,
    this.driverId,
    this.driverName,
    this.driverPhone,
    this.driverLatitude,
    this.driverLongitude,
    this.driverZone,
    this.driverVehicleType,
    this.driverUpdatedAt,
    required this.deliveryAddress,
    required this.isActive,
  });

  final int orderId;
  final String orderNumber;
  final OrderStatus status;
  final int? driverId;
  final String? driverName;
  final String? driverPhone;
  final double? driverLatitude;
  final double? driverLongitude;
  final String? driverZone;
  final String? driverVehicleType;
  final DateTime? driverUpdatedAt;
  final String deliveryAddress;
  final bool isActive;

  bool get hasDriverLocation =>
      driverLatitude != null && driverLongitude != null;

  factory OrderTracking.fromJson(Map<String, dynamic> j) => OrderTracking(
        orderId: (j['order_id'] as num).toInt(),
        orderNumber: j['order_number'] as String? ?? '',
        status: OrderStatus.fromApi(j['status'] as String?),
        driverId: (j['driver_id'] as num?)?.toInt(),
        driverName: j['driver_name'] as String?,
        driverPhone: j['driver_phone'] as String?,
        driverLatitude: (j['driver_latitude'] as num?)?.toDouble(),
        driverLongitude: (j['driver_longitude'] as num?)?.toDouble(),
        driverZone: j['driver_zone'] as String?,
        driverVehicleType: j['driver_vehicle_type'] as String?,
        driverUpdatedAt: Fmt.parseUtc(j['driver_updated_at'] as String?),
        deliveryAddress: j['delivery_address'] as String? ?? '',
        isActive: j['is_active'] as bool? ?? false,
      );
}
